frontend/src/hooks/useRouterFilter.ts

```diff
@@ -68,11 +86,11 @@ export const useRouterFilter = <T extends object>({
       // Check if query contains filter params.
       const filterProps = Object.keys(filter);
       if (_.intersection(Object.keys(params), filterProps).length) {
-        let merged = extractProps(filterProps, params);
+        let merged = { ...defaultFilter, ...extractProps(filterProps, params) };
```

Comment:
sometimes the merged object will contain a subset of all of the filter fields, this would cause a false positive on the changed flag because:
```
{searchBy: address} != {searchBy:address, agency: "", administrativeArea: "", ...}
so merge {searchBy: address} with a full list of the default, empty filter fields to ensure the above doesn't happen.